### PR TITLE
[WIP] - Add LDAP Account Form

### PR DESCRIPTION
**Pull request by utchoang**
@rhtyd cc @svenvogel 
This is my new design of the Add LDAP Account Form. Please review and let me know your comments.
Untitled

### CODE REVIEW

File: src/components/view/ActionButton.vue [hunk 1]
```diff
@@ -44,7 +44,7 @@
         v-if="action.api in $store.getters.apis &&
           !action.showBadge &&
           ((!dataView && (action.listView || action.groupAction && selectedRowKeys.length > 0)) || (dataView && action.dataView)) &&
-          ('show' in action ? action.show(resource, $store.getters.userInfo) : true)"
+          ('show' in action ? action.show(resource, $store.getters.userInfo, $store.getters.isLdapEnabled) : true)"
```
Comment (rhtyd) [inline src/components/view/ActionButton.vue]: I'm thinking to get rid of this at this.$store must be accessible by all parts of the code; or simply only pass $store.

**Comment by rhtyd** [comment]
@utchoang is this ready for testing and reviewing?

**Comment by utchoang** [comment]
@rhtyd Yes. Please review and let me know your comment

**Comment by rhtyd** [comment]
fantastic @utchoang I'll review and test and get back to you.

**Comment by svenvogel** [comment]
@utchoang thanks! cool! is is possible to add a search filter for the user? if you have many user you will find it easier...

**Comment by utchoang** [comment]
@svenvogel OK. I understood. 
@rhtyd I'll add new search filter and pagination for list users and you can review again...

**Comment by utchoang** [comment]
@svenvogel According to http://cloudstack.apache.org/api/apidocs-4.13/apis/listLdapUsers.html. I did try to add search filter, however, this API does not support keyword filter or pagination. Thus, it is impossible to add search filter in this case. I have an alternative which is to filter on client only, not by API. What do you think?

**Comment by rhtyd** [comment]
@utchoang you're right, you may refer to old code here:

https://github.com/apache/cloudstack/blob/5ff932eb868e88f298b2fb54af8bbcc688a4cc29/ui/scripts/ui-custom/accountsWizard.js#L137

**Comment by utchoang** [comment]
@rhtyd @svenvogel I added the search filter section in the Ldap users list. Please review and let me know the comments
Untitled

**Comment by rhtyd** [comment]
@utchoang thanks, will review later today/tomorrow and get back to you.

**Comment by rhtyd** [comment]
@utchoang I started review it, looks like you still have git merge/branching issues. Did you follow the previously advised steps to fix your branch?

**Comment by utchoang** [comment]
@rhtyd I realized that the master branch is now changed compared to the moment I got it earlier. Do you mean that I should get the master once again?

**Comment by rhtyd** [comment]
@utchoang the base branch you're checking out is not consistent with origin/master; i.e. your master is not same as origin/master. Kindly do this to fix it:
```
git fetch origin
git checkout master
git reset --hard origin/master
<then checkout any new branch for a new PR from your own local master, for example:>
git checkout -b feature/xyz...
```

**Comment by davidjumani** [comment]
@utchoang is there a reason the fetch data functions were all returning a promise and not used individually?

**Comment by utchoang** [comment]
@davidjumani No problem. It's that I organize my code. If you want, you can separate it out and used individually. Thanks.

**Pull request merged**